data/transforms.py

```diff
@@ -1,130 +1,156 @@
-        self.flip = TF.vflip
-        self.p = p
-
-    def __call__(self, data):
-        if np.random.rand() > self.p:
-            return data
-
-        return {task: self.flip(im) for task, im in data.items()}
-
-
-class ToTensor:
-    def __init__(self):
-        self.ImagetoTensor = trsfs.ToTensor()
-        self.MaptoTensor = self.ImagetoTensor
-
-    def __call__(self, data):
-        new_data = {}
-        for task, im in data.items():
-            if task in {"x", "a", "rx", "d", "rd"}:
-                new_data[task] = self.ImagetoTensor(im)
-            elif task in {"h", "w", "m", "rm"}:
-                new_data[task] = self.MaptoTensor(im)
-            elif task == "s":
-                new_data[task] = torch.squeeze(torch.from_numpy(np.array(im))).to(
-                    torch.int64
-                )
-            else:
-                print("ERROR: Task not found")
-        return new_data
-
-
-class Normalize:
-    def __init__(self):
-        self.normImage = trsfs.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
-        # self.normSeg = trsfs.Normalize([0.5, 0.5, 0.5], [0.5, 0.5, 0.5])
-        # self.normDepth = trsfs.Normalize([1 / 255], [1 / 3])
-        self.normMask = lambda m: (m > ((torch.max(m) - torch.min(m)) / 2.0)).float()
-
-        self.normalize = {
-            "x": self.normImage,
-            # "s": self.normSeg,
-            #"d": self.normDepth,
-            "m": self.normMask,
-            "rx": self.normImage,
-            "rm": self.normMask,
-        }
-
-    def __call__(self, data):
-        return {
-            task: self.normalize.get(task, lambda x: x)(tensor)
-            for task, tensor in data.items()
-        }
-
-
-def get_transform(transform_item):
-    """Returns the torchivion transform function associated to a
-    transform_item listed in opts.data.transforms ; transform_item is
-    an addict.Dict
-    """
-    if transform_item.name == "crop" and not transform_item.ignore:
-        return RandomCrop((transform_item.height, transform_item.width))
-
-    if transform_item.name == "resize" and not transform_item.ignore:
-        return Resize(transform_item.new_size)
-
-    if transform_item.name == "hflip" and not transform_item.ignore:
-        return RandomVerticalFlip(p=transform_item.p or 0.5)
-
-    raise ValueError("Unknown transform_item {}".format(transform_item))
-
-
-def get_transforms(opts):
-    """Get all the transform functions listed in opts.data.transforms
-    using get_transform(transform_item)
-    """
-    last_transforms = [ToTensor(), Normalize()]
-
-    conf_transforms = []
-    for t in opts.data.transforms:
-        if t.ignore:
-            continue
-
-        conf_transforms.append(get_transform(t))
-
-    return conf_transforms + last_transforms
+import torch
+from torchvision import transforms as trsfs
+import torchvision.transforms.functional as TF
+import numpy as np
+from PIL import Image
+def interpolation(task):

```

Comment:
need blank lines, flake8 can't have been happy :)

Reply:
I couldn’t use sftp this morning it wouldn’t sync my local changes to the cluster so I had to do all of this without a proper editor ...